pkgs/os-specific/linux/purism/ectool.nix

```
@@ -0,0 +1,28 @@
+{ lib, fetchgit, rustPlatform, udev, pkg-config }:
+let
+
+  gitsrc = fetchgit {
+      url = "https://source.puri.sm/firmware/librem-ec.git";
+      rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
+      sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
+    };
```

```suggestion
  gitsrc = fetchgit {
    url = "https://source.puri.sm/firmware/librem-ec.git";
    rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
    sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
  };
```
also this should use fetchFromGitLab.